src/domain/johtoselvitys/JohtoselvitysContainer.tsx

```diff
@@ -283,7 +283,12 @@ const JohtoselvitysContainer: React.FC<Props> = ({ hankeData, application }) =>
       return Promise.resolve();
     }
 
-    const mutations = newAttachments.map((file) =>
+    // Filter out attachments that have same names as those that have already been sent
+    const filesToSend = newAttachments.filter((file) =>
```

Comment:
Can this be unit tested? If so, please add a test.

Reply:
Added test.